### PR TITLE
V2 Workflows

### LINKED ISSUE #3791: 2.0 Workflows {View}
[View in Figma](https://www.figma.com/file/9MOufQNLtdkpnDucmNX10R/%E2%9D%96-Cal-DS-v2.0?node-id=13385%3A113555)

### LINKED ISSUE #3792: 2.0 Workflows / Create new {flow}
[View in Figma](https://www.figma.com/file/9MOufQNLtdkpnDucmNX10R/%E2%9D%96-Cal-DS-v2.0?node-id=13461%3A104448)

### LINKED ISSUE #3793: 2.0 Workflows Empty State {View}
[View in Figma](https://www.figma.com/file/9MOufQNLtdkpnDucmNX10R/%E2%9D%96-Cal-DS-v2.0?node-id=13385%3A113560)

### PR DESCRIPTION
**Pull request by CarinaWolli**
## What does this PR do?

This PR adds the V2 design for Workflows.

Fixes #3793
Fixes #3792
Fixes #3791

Loom Video: https://www.loom.com/share/45c9e5a6e9c448ea871eb3dbdab55201

Loom Video Mobile: https://www.loom.com/share/0f455fa3528e475bb71c2a3374cf4638

**Environment**: Staging(main branch) / Production

## How should this be tested?

- Do similar steps as shown in the loom video

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: apps/web/pages/v2/workflows/[workflow].tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { GetStaticPaths, GetStaticProps } from "next";
```
Comment (CarinaWolli) [inline apps/web/pages/v2/workflows/[workflow].tsx:1]: Just copied, no changes in this file

File: packages/features/ee/common/components/v2/LicenseRequired.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import { useSession } from "next-auth/react";
+import React, { AriaRole, ComponentType, Fragment } from "react";
+
+import { CONSOLE_URL } from "@calcom/lib/constants";
+import { Icon } from "@calcom/ui/Icon";
+import { EmptyScreen } from "@calcom/ui/v2";
```
Comment (CarinaWolli) [inline packages/features/ee/common/components/v2/LicenseRequired.tsx:6]: Copied file and added v2 import here

File: packages/features/ee/workflows/components/AddVariablesDropdown.tsx [hunk 3]
```diff
@@ -38,7 +38,7 @@ export const AddVariablesDropdown = (props: IAddVariablesDropdown) => {
               key={variable}
               type="button"
               className="px-5 py-1"
-              onClick={() => props.addVariable(props.isEmailSubject, variable)}>
+              onClick={() => props.addVariable(props.isEmailSubject, t(`${variable}_workflow`))}>
```
Comment (CarinaWolli) [inline packages/features/ee/workflows/components/AddVariablesDropdown.tsx:41]: Fixes bug that variables were always added in English and not the user's language

File: packages/features/ee/workflows/components/v2/AddActionDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,147 @@
+import { zodResolver } from "@hookform/resolvers/zod";
```
Comment (CarinaWolli) [inline packages/features/ee/workflows/components/v2/AddActionDialog.tsx:1]: Copied file and and use v2 imports

File: packages/features/ee/workflows/lib/getOptions.ts [hunk 5]
```diff
@@ -10,7 +10,9 @@ export function getWorkflowActionOptions(t: TFunction) {
 
 export function getWorkflowTriggerOptions(t: TFunction) {
   return WORKFLOW_TRIGGER_EVENTS.map((triggerEvent) => {
-    return { label: t(`${triggerEvent.toLowerCase()}_trigger`), value: triggerEvent };
+    const triggerString = t(`${triggerEvent.toLowerCase()}_trigger`);
+
+    return { label: triggerString.charAt(0).toUpperCase() + triggerString.slice(1), value: triggerEvent };
```
Comment (CarinaWolli) [inline packages/features/ee/workflows/lib/getOptions.ts:15]: Make first letter of action label uppercase

File: apps/web/pages/v2/workflows/[workflow].tsx (hunk 1 above)
Comment (CarinaWolli) [inline apps/web/pages/v2/workflows/[workflow].tsx:1]: Copied file, no changes

File: packages/trpc/server/routers/viewer/workflows.tsx [hunk 6]
```diff
@@ -126,6 +135,35 @@ export const workflowsRouter = createProtectedRouter()
       }
     },
   })
+  .mutation("createV2", {
```
Comment (CarinaWolli) [inline packages/trpc/server/routers/viewer/workflows.tsx:138]: New mutation to create workflow as we decided to remove the modal in V2 when creating a workflow

File: packages/ui/v2/core/ConfirmationDialogContent.tsx [hunk 7]
```diff
@@ -0,0 +1,76 @@
+import * as DialogPrimitive from "@radix-ui/react-dialog";
```
Comment (CarinaWolli) [inline packages/ui/v2/core/ConfirmationDialogContent.tsx:1]: Copied file and added V2 imports

**Comment by PeerRich** [comment]
exciting PR! 👏

**Comment by sean-brydon** [comment]
First of all this looks Awesome!  With help of @Jaibles we found a few UI NITs
CleanShot 2022-08-31 at 10 49 47

--- 

Our Label component should fix this issue as it provides the correct amount of spacing - would also tidy up the label styles around the page

and

For the Dialog one we actually support passing in title + description which should provide the correct amount of spacing also. (I hope :D )

**Comment by Jaibles** [comment]
Great work on this @CarinaWolli. I've left some really minor comments. Mostly they're design decisions that affect the app overall and where guidance wasn't provided - not workflow specific, so only do what makes sense in this pr from my suggestions. Thank you 🙏

**Review by ciaranha — COMMENTED** [review]
Great work on this @CarinaWolli. I've left some really minor comments in Loom. (Mostly they're design decisions that affect the app overall and where guidance wasn't provided - not workflow specific, so only do what makes sense in this pr from my suggestions.) Thank you 🙏

File: apps/web/pages/v2/workflows/[workflow].tsx (hunk 1 above)
Comment (zomars) [reply apps/web/pages/v2/workflows/[workflow].tsx:1]: Will add some deprecation comments to the v1 file

File: packages/features/ee/common/components/v2/LicenseRequired.tsx (hunk 2 above)
Comment (zomars) [reply packages/features/ee/common/components/v2/LicenseRequired.tsx:6]: Added deprecation notice

**Review by zomars — APPROVED** [review]
Awesome work @CarinaWolli Added some deprecation comments and merging after all checks pass 🙏

**Comment by CarinaWolli** [comment]
> Great work on this @CarinaWolli. I've left some really minor comments in Loom. (Mostly they're design decisions that affect the app overall and where guidance wasn't provided - not workflow specific, so only do what makes sense in this pr from my suggestions.) Thank you 🙏

Thank you for your feedback @Jaibles 🙏 Feedback is implemented here: https://github.com/calcom/cal.com/pull/4070

**Pull request merged**